pkg/daemon/ceph/osd/kms/kms.go

```diff
@@ -238,6 +238,25 @@ func (c *Config) UpdateSecret(secretName, secretValue string) error {
 
 		return nil
 	}
+	if c.IsVault() {
+		// Store the secret in Vault
+		v, err := InitVault(c.ClusterInfo.Context, c.context, c.ClusterInfo.Namespace, c.clusterSpec.Security.KeyManagementService.ConnectionDetails)
+		if err != nil {
+			return errors.Wrap(err, "failed to init vault kms")
+		}
+		k := buildVaultKeyContext(c.clusterSpec.Security.KeyManagementService.ConnectionDetails)
+		// Build Secret
+		secretName = GenerateOSDEncryptionSecretName(secretName)
+		data := make(map[string]interface{})
```

Comment:
```suggestion
data := make(map[string]string)
```
secretValue is of type string, why not use that as we know the type?

Reply:
Rook uses [libopenstorage/secrets](https://github.com/libopenstorage/secrets) library for KMS and its implementation of PutSecret is as follows
```
PutSecret(
	secretId string,
	plainText map[string]interface{},
	keyContext map[string]string,
) (Version, error)
```